aptos-move/framework/aptos-token/sources/token.move

```diff
@@ -1475,6 +1475,34 @@ module aptos_token::token {
         token_data.royalty
     }
 
+    /// return the token_data_id from the token_id
+    public fun get_tokendata_id(token_id: TokenId): TokenDataId {
+        token_id.token_data_id
+    }
+
+    /// return the mutation setting of the tokendata
+    public fun get_tokendata_mutability_config(token_data_id: TokenDataId): (bool, bool, bool, bool, bool) acquires Collections {
+        let creator_addr = token_data_id.creator;
+        assert!(exists<Collections>(creator_addr), error::not_found(ECOLLECTIONS_NOT_PUBLISHED));
+        let all_token_data = &borrow_global<Collections>(creator_addr).token_data;
+        assert!(table::contains(all_token_data, token_data_id), error::not_found(ETOKEN_DATA_NOT_PUBLISHED));
+        let config = table::borrow(all_token_data, token_data_id).mutability_config;
+        (config.maximum, config.uri, config.royalty, config.description, config.properties)
+    }
+
+    /// return the collection mutation setting
```

Comment:
I wonder if this can be marked as #[view]. Can you try that and see if it works? View functions cannot take structs as inputs but I wonder if String still works

Reply:
added view